state/redis/redis.go

```diff
@@ -7,43 +7,50 @@ package redis
 
 import (
 	"context"
-	"encoding/json"
+	"crypto/tls"
```

Comment:
Could you please include the integration test file as @youngbupark suggested (clean it up a bit and remove the credentials from it). It would serve as a guide for other contributors. If its going to take longer then please finish this PR and add that as a separate PR after this release.

Reply:
I will add it as a separate PR but will try to hit it for this release.